plugins/inputs/vsphere/README.md

```diff
@@ -551,9 +561,231 @@ For a detailed list of commonly available metrics, please refer to
 * virtualDisk stats for VM
   * disk (name of virtual disk)
 
+## Add a vSAN extension
+
+A vSAN resource is a special type of resource that can be collected by the
+plugin. The configuration of a vSAN resource slightly differs from the
+configuration of hosts, VMs, and other resources.
+
+### Prerequisites for vSAN
+
+* vSphere 6.5 and later
+* Clusters with vSAN enabled
+* [Turn on Virtual SAN performance service](https://docs.vmware.com/en/VMware-vSphere/6.5/com.vmware.vsphere.virtualsan.doc/GUID-02F67DC3-3D5A-48A4-A445-D2BD6AF2862C.html): When you create a vSAN cluster,
+the performance service is disabled. To monitor the performance metrics,
+you must turn on vSAN performance service.
+
+### vSAN Configuration
+
+```toml
+[[inputs.vsphere]]
+  interval = "300s"
+  vcenters = ["https://<vcenter-ip>/sdk", "https://<vcenter2-ip>/sdk"]
+  username = "<user>"
+  password = "<pwd>"
+
+  # Exclude all other metrics
+  vm_metric_exclude = ["*"]
+  datastore_metric_exclude = ["*"]
+  datacenter_metric_exclude = ["*"]
+  host_metric_exclude = ["*"]
+  cluster_metric_exclude = ["*"]
+  
+  # By default all supported entity will be included
+  vsan_metric_include = [
+    "summary.disk-usage",
+    "summary.health",
+    "summary.resync",
+    "performance.cluster-domclient",
+    "performance.cluster-domcompmgr",
+    "performance.host-domclient",
+    "performance.host-domcompmgr",
+    "performance.cache-disk",
+    "performance.disk-group",
+    "performance.capacity-disk",
+    "performance.disk-group",
+    "performance.virtual-machine",
+    "performance.vscsi",
+    "performance.virtual-disk",
+    "performance.vsan-host-net",
+    "performance.vsan-vnic-net",
+    "performance.vsan-pnic-net",
+    "performance.vsan-iscsi-host",
+    "performance.vsan-iscsi-target",
+    "performance.vsan-iscsi-lun",
+    "performance.lsom-world-cpu",
+    "performance.nic-world-cpu",
+    "performance.dom-world-cpu",
+    "performance.cmmds-world-cpu",
+    "performance.host-cpu",
+    "performance.host-domowner",
+    "performance.host-memory-slab",
+    "performance.host-memory-heap",
+    "performance.system-mem",
+  ]
+  # by default vsan_metric_skip_verify = false
+  vsan_metric_skip_verify = true
+  # vsan_metric_exclude = ["*"]
```

Comment:
So all the mentioned metrics above will be excluded? (as `*` is the default)

Reply:
Yes, the user can enable these metrics on the requirement. It depends on the environment.

Reply:
But they will all be excluded because of the *. The example config is wrong IMHO.

Reply:
> So all the mentioned metrics above will be excluded? (as `*` is the default)

Excludes metrics only when `vsan_metric_exclude = ["*"]` is uncommented. By default `vsan_metric_include` is used to collect the metrics.

Reply:
The example config should show the defaults when commented. The example config currently shows * is the default for metric_exclude, which means all metrics get excluded.

If that is not the case, the sample config should show this:
```toml
# vsan_metric_exclude = []
```

Reply:
> The example config currently shows * is the default for metric_exclude, which means all metrics get excluded.

That is correct, we want vsan metrics to be opt-in. This is similar to what you will see for `datacenter_metric_*` and `custom_attribute_*`

Reply:
But as soon as you put some values in the include, the default value of exclude isn't `*` anymore?

Reply:
Please point to code you are concerned about.

Reply:
I did 🤷

Reply:
And we have concluded that this sample config options is consistent with the other metrics this plugin collects.

Reply:
So the following config will load the summary health?
```toml
  vsan_metric_include = ["summary.health"]
# vsan_metric_exclude = ["*"]
```
What about this one?
```toml
  vsan_metric_include = ["summary.health"]
  vsan_metric_exclude = ["*"]
```
(include summary, but exclude all?)

Reply:
@Hipska You are right, this should be the sample config.

```
  vsan_metric_include = []
  # by default vsan_metric_skip_verify = false
  vsan_metric_skip_verify = true
  # vsan_metric_exclude = [ ]
  vsan_cluster_include = [ "/*/host/**" ]
```

The config
```
  vsan_metric_include = ["summary.health"]
# vsan_metric_exclude = [ ]
```
OR
```
  vsan_metric_include = ["summary.health"]
  vsan_metric_exclude = [ ]
```
collects `summary.health` metric.